Core/NoOpRunner.Core/Shapes/EntityShape.cs

```diff
@@ -0,0 +1,13 @@
+﻿using NoOpRunner.Core.Entities;
+using System.Diagnostics;
+
+namespace NoOpRunner.Core.Shapes
+{
+    public abstract class EntityShape : BaseShape
```

Comment:
About what to put into these classes. For starters, I was thinking something along the lines of: 
`BaseShape` should have two abstract methods: `bool CanOverlap(BaseShape other)`, `void OnCollision(BaseShape other)`. Child classes override these methods.
For passable shapes (damageCrystals, healthCrystals, passablePlatforms), CanOverlap implementation should return true (maybe with extra movement direction checks for passablePlatforms, not sure about exact impl, can be stubbed for now), OnCollision should triger the appropriate effects (can be stubbed for now).
For impassable shapes, CanOverlap should return false, OnCollision should do damage for saws, explode for rockets, do nothing (or maybe stop movement, exact impl unsure) for platforms.
Then, in theory, we can handle all the shapes and their collisions without explicitly caring about what shapes they are.
But this is only the general direction, you should polish the idea while implementing it, change where need be

Reply:
Implemented empty functions for now.